api/net/Areas/Editor/Controllers/StorageController.cs

```diff
@@ -49,216 +55,627 @@ public StorageController(IOptions<StorageOptions> options)
+                    var safePath = Path.Combine(locationPath, path);
+                    if (!client.Exists(safePath)) throw new InvalidOperationException($"Stream does not exist: '{path}'");
 
-        var info = new ItemModel(safePath);
-        var stream = System.IO.File.OpenRead(safePath);
-        return File(stream, contentType: info.MimeType!, fileDownloadName: info.Name, enableRangeProcessing: true);
+                    var info = new ItemModel(Path.GetFileName(safePath), client.GetAttributes(safePath));
+                    var tmpPath = _connection.CopyFile(client, safePath, _options.GetCapturePath(), this.User);
+                    var stream = System.IO.File.OpenRead(tmpPath);
+                    return File(stream, contentType: info.MimeType!, fileDownloadName: info.Name, enableRangeProcessing: false);
+                }
+                finally
+                {
+                    if (client.IsConnected)
+                        client.Disconnect();
+                }
+            }
+            else if (connection == null || connection?.ConnectionType == ConnectionType.LocalVolume)
+            {
+                var safePath = Path.Combine(_options.GetCapturePath(), path);
+                if (!safePath.FileExists()) throw new InvalidOperationException($"Stream does not exist: '{path}'");
+
+                var info = new ItemModel(safePath);
+                var stream = System.IO.File.OpenRead(safePath);
+                return File(stream, contentType: info.MimeType!, fileDownloadName: info.Name, enableRangeProcessing: true);
+            }
+            else throw new NotImplementedException($"Location connection type '{connection?.ConnectionType}' not implemented yet.");
+        }
+        else
+        {
+            var safePath = Path.Combine(_options.GetUploadPath(), path);
+            if (!safePath.FileExists()) throw new InvalidOperationException($"Stream does not exist: '{path}'");
+
+            var info = new ItemModel(safePath);
+            var stream = System.IO.File.OpenRead(safePath);
+            return File(stream, contentType: info.MimeType!, fileDownloadName: info.Name, enableRangeProcessing: true);
+        }
     }
 
     /// <summary>
     /// Download the file for the specified path.
     /// </summary>
+    /// <param name="locationId"></param>
     /// <param name="path"></param>
-    /// <param name="location"></param>
     /// <returns></returns>
     [HttpGet("download")]
-    [HttpGet("{location}/download")]
+    [HttpGet("{locationId:int}/download")]
     [Produces("application/octet-stream")]
     [ProducesResponseType(typeof(FileStreamResult), (int)HttpStatusCode.OK)]
     [ProducesResponseType((int)HttpStatusCode.BadRequest)]
     [SwaggerOperation(Tags = new[] { "Storage" })]
-    public IActionResult Download(string path, [FromRoute] string? location = "capture")
+    public IActionResult Download([FromRoute] int? locationId, [FromQuery] string path)
```

Comment:
I wish we could download directly from the remote location, however it loses the connection.  The only solution so far is to copy it locally first.